File: 398/random-pick-index.py
import random


class Solution():

    def __init__(self, nums):
        """
        :type nums: List[int]
        :type numsSize: int
        """
        self.nums = nums

    def pick(self, target):
        """
        :type target: int
        :rtype: int
        """
        res = []
        for index, item in enumerate(self.nums):
            if item == target:
                res.append(index)

        index = int(random.random() * len(res))
        return res[index]

File: 398/RandomPickIndex.scala
object RandomPickIndex {
  class Solution(_nums: Array[Int]) {

    def pick(target: Int): Int =
      _nums.foldLeft((-1, -1, 0)) { case ((pickIndex, count, i), num) =>
          if (num == target) {
            if (util.Random.nextInt(count) == 0) (i, count + 1, i + 1)
            else (pickIndex, count + 1, i + 1)
          }
          else (pickIndex, count + 1, i + 1)
      }._1

  }

  /**
    * Your Solution object will be instantiated and called as such:
    * var obj = new Solution(nums)
    * var param_1 = obj.pick(target)
    */
  def main(args: Array[String]): Unit = {
    var obj = new Solution(Array(1,2,3,3,3))
    var param_1 = obj.pick(3)
    println(param_1)
  }
}
